packages/visual-utils/src/calculateSegments.ts

```diff
@@ -77,20 +76,29 @@ export default function calculateSegments(
 			.range([startColor as any, endColor as any])
 	}
 
-	return _.sortBy(segmentInfo, ['name']).map((v, i) => {
-		let color = fullColors[i]
-		if (segmentColors && segmentColors[i]) {
-			color = segmentColors[i].color
-		} else if (gradientScale) {
-			color = gradientScale(isNaN(v.name) ? i : v.name) as any
-		} else {
-			color = defaultColor || fullColors[i]
-		}
-		color = color || '#ccc'
-		return {
-			name: v.name,
-			identity: v.identity,
-			color
-		}
-	})
+	return segmentInfo
+		.sort((a, b) => {
+			if ((!a && b) || a.name < b.name) {
+				return -1
+			} else if ((a && !b) || a.name > b.name) {
+				return 1
+			}
+			return 0
+		})
+		.map((v, i) => {
```

Comment:
If you destructure name and identity here, you can use object-shorthand in the return later